Make color scheme less clinical

mtdowling opened the pull request:
This commit updates the website color scheme to be a brownish red / iron
ore color scheme rather than the more clinical feeling blue. Several
other theme changes are made, including putting borders above headings
rather than below, automatically numbering headings in specification
pages using CSS, and adding contrast between the nav sidebar and the
content.

Homepage:

![Screen Shot 2020-03-30 at 10 58 33 AM](https://user-images.githubusercontent.com/190930/77945408-75b8b200-7275-11ea-9602-b47c9fec5955.png)

Top of content pages:

![Screen Shot 2020-03-30 at 11 01 36 AM](https://user-images.githubusercontent.com/190930/77945716-f11a6380-7275-11ea-8484-930a4b3e5ae0.png)

Inside of content pages:

![Screen Shot 2020-03-30 at 10 59 17 AM](https://user-images.githubusercontent.com/190930/77945443-849f6480-7275-11ea-915f-0df8fd8dc03f.png)

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

kstich commented:
In general, I like the scheme a lot better. My only comment is that there appears to be no visual separation between the persistent navbar and the front page's "banner" area, color or otherwise.

mtdowling commented:
Updated with more visual distinction between the header and the splash thingy. I updated the tag line too to be less wordy... What do you think?

![Screen Shot 2020-03-30 at 11 29 28 AM](https://user-images.githubusercontent.com/190930/77948126-bfa39700-7279-11ea-8f35-e90e4c0be804.png)

jgalperin commented:
Looking good! To my eye, I think the difference in color is too contrasting. Maybe we could use something like [this tool](https://coolors.co/fa7921-fe9920-7f504b-b9a44c-0c4767) to find complimentary colors that are still distinct enough?

mtdowling commented:
How about this then?

![Screen Shot 2020-03-30 at 11 50 23 AM](https://user-images.githubusercontent.com/190930/77949951-abad6480-727c-11ea-9779-26712abe5528.png)
